jupyterlab_bigquery/src/utils/widget_manager.ts

```diff
@@ -9,29 +9,46 @@ import { ReduxReactWidget } from './ReduxReactWidget';
  * A class that manages dataset widget instances in the Main area
  */
 export class WidgetManager {
+  private static instance: WidgetManager = undefined;
   private widgets: { [id: string]: Widget } = {};
   private reduxWidgets: { [id: string]: ReduxReactWidget } = {};
   private store: EnhancedStore;
 
-  constructor(private app: JupyterFrontEnd) {
+  private constructor(private app: JupyterFrontEnd) {
     this.store = configureStore({ reducer: rootReducer });
 
     return this;
   }
 
+  static initInstance(app?: JupyterFrontEnd) {
```

Comment:
Can the `?` be removed, and make this a required parameter and remove the `app !== undefined`